op-node/rollup/interop/managed/system.go

```diff
@@ -328,11 +330,14 @@ func (m *ManagedMode) Reset(ctx context.Context, lUnsafe, xUnsafe, lSafe, xSafe,
 		return err
 	}
 
+	latestLocalUnsafe, err := m.scanL2ForLatestLocalUnsafe(ctx, lUnsafe)
```

Comment:
May I ask why do we not directly use `lUnsafe`? So to _save_ more unsafe l2 blocks, not to reorg? Since this `lUnsafe` is given by supervisor, can we safely trust that this is a best effort and directly use it?

Reply:
May I ask this behavior is consistent to non-interop op-node? My understanding of the PR is when L1 reorg, make sure L2 unsafe heads reorg with best effort to save unsafe L2 block as much as possible. So, how does pre-interop does it? Does it also go for best effort? If yes, how did it approach?

Reply:
On `develop` the behavior is not consistent with non-interop. This PR makes the behavior consistent.

> when L1 reorg, make sure L2 unsafe heads reorg with best effort to save unsafe L2 block as much as possible
Prior to this PR, L2 unsafe chain stays invalid after a L1 reorg, it features blocks that contain reorged L1Origin fields, and need to be removed.

With this PR, we update the unsafe chain and remove all invalid blocks, but not more than necessary (i.e. we don't remove valid blocks).

> how does pre-interop does it?
Codepath for pre-interop is totally different - we don't use `ManagedMode`. Pre-interop, we have `SyncDeriver.onResetEvent`  which is short-circuited for managed mode.
In a non-interop network this behavior/logic is handled in `FindL2Heads`.